### PR TITLE
Cleanup record access

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
- Removes a vestigial code path in `GetTxnRecordAnswer` that looks for the the `TransactionRecord` in the `TransactionID`'s payer account when it is missing from the `txnHistories` map.
    * After we stopped creating 25-hour threshold records, the payer account will have the record iff it is still in the `txnHistories` map. (Both live for 180 seconds, and are removed simultaneously during record expiration.)
- Limits the payer records returned for a `CryptoGetAccountRecordsQuery` by the global/dynamic property `ledger.records.maxQueryableByAccount=180`.
    * Avoids retrieving these records twice (once for fee calculation, once for query response) by using the `Map<String, Object> queryCtx` set [by the `StakedAnswerFlow`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/queries/answering/StakedAnswerFlow.java#L127).

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Comment by rbair23** [comment]
I'm taking a deeper look at the PR. If we are going to add a mechanism to limit the number of returned records (and we really do need this), then we should also talk about pagination or watermarking. Since the number of transactions change all the time, I don't think we can do pagination (it would be basically worthless). Instead we should allow the user to say "get me the next N number of records newer than time T" or "newer than record R", or something along those lines. And we have a cap (as proposed in this PR) for the number of records returned. We can do that as a second PR, but we must recognize that this PR does introduce a breaking change in our API behavior and we need to make sure we document it as such as well.

**Comment by tinker-michaelj** [comment]
> I'm taking a deeper look at the PR. If we are going to add a mechanism to limit the number of returned records (and we really do need this), then we should also talk about pagination or watermarking. Since the number of transactions change all the time, I don't think we can do pagination (it would be basically worthless). Instead we should allow the user to say "get me the next N number of records newer than time T" or "newer than record R", or something along those lines. And we have a cap (as proposed in this PR) for the number of records returned. We can do that as a second PR, but we must recognize that this PR does introduce a breaking change in our API behavior and we need to make sure we document it as such as well.

Created [a PR](https://github.com/hashgraph/hedera-protobufs/pull/144) in _hedera-protobufs_ to support this watermarking.

File: hedera-node/src/main/java/com/hedera/services/queries/answering/AnswerFunctions.java [hunk 1]
```diff
@@ -21,54 +21,110 @@
...
+	private List<TransactionRecord> mostRecentFrom(final MerkleAccount account, final int m, final int n) {
+		final List<TransactionRecord> ans = new ArrayList<>();
+		final Iterator<ExpirableTxnRecord> iter = account.recordIterator();
+		try {
+			for (int i = 0, cutoff = n - m; i < n; i++) {
+				final var nextRecord = iter.next();
+				if (i >= cutoff) {
+					ans.add(nextRecord.asGrpc());
+				}
 			}
```
Comment (rbair23) [inline hedera-node/src/main/java/com/hedera/services/queries/answering/AnswerFunctions.java:124]: Ouch, this looks inefficient. It looks like the platform could easily add a reverse iterator, which would keep you from having to iterate over the entire body of records in this case.

File: hedera-node/src/main/java/com/hedera/services/queries/crypto/GetAccountRecordsAnswer.java [hunk 2]
```diff
@@ -112,6 +153,6 @@ public HederaFunctionality canonicalFunction() {
 	@Override
 	public Optional<SignedTxnAccessor> extractPaymentFrom(final Query query) {
 		final var paymentTxn = query.getCryptoGetAccountRecords().getHeader().getPayment();
-		return Optional.ofNullable(SignedTxnAccessor.uncheckedFrom(paymentTxn));
+		return Optional.of(SignedTxnAccessor.uncheckedFrom(paymentTxn));
```
Comment (rbair23) [inline hedera-node/src/main/java/com/hedera/services/queries/crypto/GetAccountRecordsAnswer.java:156]: What was the significance of this change?

File: hedera-node/src/main/java/com/hedera/services/queries/answering/AnswerFunctions.java (hunk 1 above)
Comment (rbair23) [reply hedera-node/src/main/java/com/hedera/services/queries/answering/AnswerFunctions.java:124]: I filed an issue with Swirlds. We can come back and look at making this more efficient after reverse iteration is supported.

**Review by rbair23 — APPROVED** [review]

File: hedera-node/src/main/java/com/hedera/services/queries/crypto/GetAccountRecordsAnswer.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/queries/crypto/GetAccountRecordsAnswer.java:156]: The gRPC accessors never return `null`, so IntelliJ was marking this `Optional.ofNullable()` as a code smell.

**Pull request merged**